### PR TITLE
More docs and different layout

### PR DESCRIPTION
**Pull request by masci**
### What does this PR do?

Adds more docs, specifically a part dedicated to listing what the Agent6 is missing to get feature parity with version 5.

Beta docs were isolated into a single document while important info that will be still valid after we exit the Beta were moved to a dedicated section for the Agent.

A stub for documenting how standalone DSD works was added.

TOCs were added as well.

### Motivation

Support request.

### Additional Notes

To ease reviewing, I suggest to navigate https://github.com/DataDog/datadog-agent/tree/massi/docs/docs more than looking at the diff.

### CODE REVIEW

**Comment by xvello** [comment]
The `known issues` link in [the jmx section](https://github.com/DataDog/datadog-agent/blob/massi/docs/docs/agent/changes.md#jmx) is a dead link, but it seems like it already was on master

**Review by xvello — DISMISSED** [review]
LGTM 🛒

File: docs/agent/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * [Upgrade][upgrade]
+
+## What is Agent 6?
+
+Agent 6 is the latest major version of the Datadog Agent. The big difference
+between Agent 5 and Agent 6 is that Agent 6 is a complete rewrite of the core
+Agent in Golang. Don’t worry! We still support Python checks.
+
+## Installation and management
+
```
Comment (olivielpeau) [inline docs/agent/README.md]: To me it'd make sense to add a reminder here that the agent 6 is currently in Beta, and link to the beta doc page

File: docs/dev/README.md [hunk 2]
```diff
@@ -2,6 +2,19 @@
 
 This guide will help you to develop and contribute code to the project.
 
+## TOC
+
+ * [Checks developer guide](checks/README.md)
+   * [Python checks](checks/python/README.md)
```
Comment (olivielpeau) [inline docs/dev/README.md]: this link doesn't work, there's no README under `checks/python`

**Review by olivielpeau — DISMISSED** [review]
Looks great, just a couple of nits 👌

File: docs/agent/gui.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Graphical User Interface
+
+Agent 6 deprecated Agent5's Windows Agent Manager GUI, replacing it with a browser-based, cross-platform one.
+
+## Using the GUI
+
+The port which the GUI runs on can be configured in your  `datadog.yaml` file. Setting the port to -1 disables the GUI all together. By default it is enabled on port `5002` on Windows and Mac, and is disabled on Linux.
```
Comment (mfpierre) [inline docs/agent/gui.md]: your  `datadog.yaml` double space

File: docs/agent/gui.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+Once the Agent is running, use the `datadog-agent launch-gui` command to launch the GUI within your default web browser.
+
+
+## Requirements
+
+1. Cookies must be enabled in your browser. The GUI generates and saves a token in your browser which is used for authenticating all communications with the GUI server.
+
+2. The GUI will only be launched if the user launching it has the correct user permissions: if you are able to open `datadog.yaml`, you are able to use the GUI.
+
+3. For security reasons, the GUI can **only** be accessed from the local network interface (```localhost```/```127.0.0.1```), so you must be on the same host that the agent is running to use it. In other words, you can't run the agent on a VM and access it from the host machine.
```
Comment (mfpierre) [inline docs/agent/gui.md]: I guess we could add "on a VM or a container"

File: docs/dev/agent_api.md [hunk 5]
```diff
@@ -1,10 +1,16 @@
...
-and [Named Pipes](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365590.aspx)
-on Windows, authorization is delegated to the filesystem.
+development of 3rd party tools and interfaces. By default, the API is available
+at TCP port `5001` on `localhost` but can be configured differently.
+
+## Security and Authentication
+
+To avoid unprivileged users to access the API, authentication is required.
+
+FIXME: new authentication system wasn't merged yet: https://github.com/DataDog/datadog-agent/pull/939
```
Comment (olivielpeau) [inline docs/dev/agent_api.md]: fyi it's merged now, will be in `beta.7`

**Review by olivielpeau — APPROVED** [review]
one small note, other than that LGTM, feel free to merge 🚀

File: docs/dev/agent_api.md (hunk 5 above)
Comment (masci) [reply docs/dev/agent_api.md]: Merging for now, that would be a fair large addition to the docs

**Pull request merged**